Userland/Libraries/LibDeviceTree/DeviceTree.cpp

```diff
@@ -93,4 +94,173 @@ bool Node::is_compatible_with(StringView wanted_compatible_string) const
     return is_compatible;
 }
 
+ErrorOr<Reg> Node::reg() const
+{
+    if (parent() == nullptr)
+        return Error::from_errno(EINVAL);
+
+    auto reg_prop = get_property("reg"sv);
+    if (!reg_prop.has_value())
+        return Error::from_errno(ENOENT);
+
+    return Reg { reg_prop->raw_data, *this };
+}
+
+ErrorOr<Ranges> Node::ranges() const
+{
+    if (parent() == nullptr)
+        return Error::from_errno(EINVAL);
+
+    auto ranges_prop = get_property("ranges"sv);
+    if (!ranges_prop.has_value())
+        return Error::from_errno(ENOENT);
+
+    return Ranges { ranges_prop->raw_data, *this };
+}
+
+ErrorOr<Address> RegEntry::resolve_root_address() const
+{
+    VERIFY(m_node.parent() != nullptr);
+    return m_node.parent()->translate_child_bus_address_to_root_address(bus_address());
+}
+
+ErrorOr<RegEntry> Reg::entry(size_t index) const
+{
+    if (index >= entry_count())
+        return Error::from_errno(EINVAL);
+
+    VERIFY(m_node.parent() != nullptr);
+
+    auto parent_address_cells = m_node.parent()->address_cells();
+    auto parent_size_cells = m_node.parent()->size_cells();
+
+    size_t const start_index = index * ((parent_address_cells + parent_size_cells) * sizeof(u32));
```

Comment:
nit: redundant parens around the right multiplication

Reply:
(in some other places clang-tidy got angry at me for using multiplication without parentheses and just using normal operator precedence rules)